### PR TITLE
Can open any URL with baseMediaProvider, navigate between youtube videos

### PR DESCRIPTION
**Pull request by jfstn**
Solves #52

### CODE REVIEW

**Comment by poush** [comment]
@jfstn I think you missed the original issue here. This is related to #9 and the PR of it. We currently blocks all external links and show an alert with instructions to open the url on any browser. Instead, we can open that url on the same window by preventing new window and calling providers (as when cmd/ctrl + shift + v) is pressed.

**Comment by jfstn** [comment]
Hello, I'll take care of it after work.

**Comment by jfstn** [comment]
```	
       get getContent() {
		let link = clipboard.readText('selection')
		if(this.extractContents(link)){
			// console.log(this.response)
			return this
		}

		throw "InvalidContentByContentExtraction"
	}
```

Wouldn't it be better to change this *getter* to not rely on `clipboard.readText('selection')`?

Made a new commit. Check it out when you can.

File: ServiceProviders/providers.js [hunk 1]
```diff
@@ -9,14 +9,20 @@ let matchers = {
...
+	
 }
 
 module.exports = {
 
 	run(win) {
 
-		let text = clipboard.readText('selection')
+		if(arguments[1]) {
+			clipboard.writeText(arguments[1])
```
Comment (poush) [inline ServiceProviders/providers.js]: I think we should writing anything to the clipboard. This will replace anything already present on Clipboard.

**Comment by poush** [comment]
@jfstn Agree, the design of baseMediaProvider is not accurate right now. This getter should not depend upon clipboard, instead, must be provided with the link as an argument or object member.

**Comment by jfstn** [comment]
@poush getters can't be used with arguments. Instead a setter should be used. That's what I did, pushing at the moment.

**Comment by poush** [comment]
This pull request **introduces 1 alert** when merging b7235782572933561598a52da3ed42ea102f4ad1 into 54e54ec15eb8426d7170e1300925249b9bddc882 - [view on LGTM.com](https://lgtm.com/projects/g/poush/h2/rev/pr-569dbb17ac9b38ba714c48a854065eb5e50de27a)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**